src/gui/transferlistsortmodel.cpp

```diff
@@ -39,6 +39,44 @@
 
 namespace
 {
+    bool operator<(const QVariant &left, const QVariant &right)
+    {
+        if (left.userType() == QMetaType::UnknownType)
+            return false;
+
+        if (right.userType() == QMetaType::UnknownType)
+            return true;
+
+        switch (left.userType())
+        {
+        case QMetaType::Bool:
+            return left.toBool() < right.toBool();
+        case QMetaType::Int:
+            return left.toInt() < right.toInt();
+        case QMetaType::UInt:
+            return left.toUInt() < right.toUInt();
+        case QMetaType::LongLong:
+            return left.toLongLong() < right.toLongLong();
+        case QMetaType::ULongLong:
+            return left.toULongLong() < right.toULongLong();
+        case QMetaType::Float:
+            return left.toFloat() < right.toFloat();
+        case QMetaType::Double:
+            return left.toDouble() < right.toDouble();
+        case QMetaType::QChar:
+            return left.toChar() < right.toChar();
+        case QMetaType::QDate:
+            return left.toDate() < right.toDate();
+        case QMetaType::QTime:
+            return left.toTime() < right.toTime();
+        case QMetaType::QDateTime:
+            return left.toDateTime() < right.toDateTime();
+        case QMetaType::QString:
+        default:
```

Comment:
Maybe it is better to `assert(false);` for `default` case. QVariant can have container types in it.

Reply:
>maybe also add an assert that will check left and right have the same userType.

>Maybe it is better to assert(false); for default case. QVariant can have container types in it.

Done.